mantidproject/mantid
# SCD interface: proton charge if no monitors and set goniometer

**Description of work.**
If CORELLI, 6 possible goniometer angle logs are summed and the total is added as phi. If monitors are missing from NeXus file, proton charge is used instead.


**Report to:* Feng Ye* [user name]/[nobody]. 

**To test:**
Test SCD Event Data Reduction interface with /SNS/CORELLI/IPTS-20927/nexus/CORELLI_70490.nxs.h5
Interface should get error about no monitors and then use proton charge as monitor count.  Run SaveIsawPeaks using peaks workspace to look at file for MONCNT and goniometer angles.



Fixes #24266 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
